lib/install/mutate-into-logical-tree.js

```diff
@@ -70,18 +70,27 @@ function translateTree_ (tree, seen) {
   tree.children.forEach(function (child) {
     pkg.dependencies[moduleName(child)] = translateTree_(child, seen)
   })
-  Object.keys(tree.missingDeps).forEach(function (name) {
+
+  function markMissing (name, requiredBy) {
     if (pkg.dependencies[name]) {
+      if (pkg.dependencies[name].missing) return
```

Comment:
Forgive me for a question to which I feel I should already know the answer, but does the logical tree conflate `dependencies` and `devDependencies`? I get why there are two passes for `tree.missingDeps` and `tree.missingDevDeps` now, but I find both being flattened down into a single dependency list here confusing.